extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/array/IntArrayExpansionKernel.java

```diff
@@ -100,28 +101,22 @@ public <A extends Any> WritableObjectChunk<int[], A> contract(
             @Nullable final WritableChunk<A> outChunk,
             final int outOffset,
             final int totalRows) {
-        if (lengths != null && lengths.size() == 0
-                || lengths == null && offsets != null && offsets.size() <= 1) {
-            if (outChunk != null) {
-                return outChunk.asWritableObjectChunk();
-            }
-            final WritableObjectChunk<int[], A> chunk = WritableObjectChunk.makeWritableChunk(totalRows);
-            chunk.fillWithNullValue(0, totalRows);
-            return chunk;
-        }
-
         sizePerElement = Math.abs(sizePerElement);
         final int itemsInBatch = offsets == null
                 ? source.size() / sizePerElement
                 : (offsets.size() - (lengths == null ? 1 : 0));
         final IntChunk<A> typedSource = source.asIntChunk();
-        final WritableObjectChunk<int[], A> result;
-        if (outChunk != null) {
-            result = outChunk.asWritableObjectChunk();
-        } else {
-            final int numRows = Math.max(itemsInBatch, totalRows);
-            result = WritableObjectChunk.makeWritableChunk(numRows);
-            result.setSize(numRows);
+        final WritableObjectChunk<int[], A> result = BaseChunkReader.castOrCreateChunk(
+                outChunk,
+                outOffset,
+                Math.max(totalRows, itemsInBatch),
+                WritableObjectChunk::makeWritableChunk,
+                WritableChunk::asWritableObjectChunk);
+
+        if (lengths != null && lengths.size() == 0
```

Comment:
What tests are hitting this new logic?
![image](https://github.com/user-attachments/assets/f96e6dd6-575e-4622-b1be-01ef538eff3a)